### PR TITLE
[GITHUB-13] Add support for v5.0.0

### PR DESCRIPTION
**Pull request by wojtek-oledzki**
This commit adds new version 1.2 and changes default version 2.4.1

Rearranged templates and tasks to avoid ugly "if"s.

### CODE REVIEW

File: templates/10-elasticsearch.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# {{ ansible_managed }}
+# file: elasticsearch/templates/10-elasticsearch.conf
+
+{{ elasticsearch.user }}    soft    nofile    65535
```
Comment (racer1680) [inline templates/10-elasticsearch.conf]: Are you sure 65535 is enough?

**Comment by starkers** [comment]
Looks good, I like the different template per family aproach

One thing I suggest we talk about is that the `cloud-aws` plugin is now installed on 2x as:

```
 - plugin_name: cloud-aws
   plugin_dir:  cloud-aws
```

..so I guess its either a case of defaulting the plugins per family (eg slightly different per family like `HQ/hq`) or maybe just some working config examples in the README I suppose?

File: templates/10-elasticsearch.conf (hunk 1 above)
Comment (starkers) [inline templates/10-elasticsearch.conf]: Thats correct actually :-)

...if it was possible to have more that number would be elastic `best practices` :+1:

**Comment by starkers** [comment]
just thinking about `best practices`, should we take the opportunity to make sure dynamic scripting is off on default too? (is anyone that we know of using it anyway?)

**Comment by wojtek-oledzki** [comment]
@starkers 
- I will update README with plugins and versions
- I will need example config entires, it's different in each version family :(. Alternatively I can add support for entire `elasticsearch.yml` in `elsticsearch.config` object like

```YAML
# defautls/main.yml
elasticsearch:
  ...
  config:
    node.name: "{{ ansible_hostname }}"
    cluster.name: my-cluster
    path.data: /var/lib/elasticsearch
    path.logs: /var/log/elasticsearch
    ...
```
This way you can control the config file, and add any custom settings from you playbook/var file.

File: templates/10-elasticsearch.conf (hunk 1 above)
Comment (wojtek-oledzki) [inline templates/10-elasticsearch.conf]: I can move it to config.

No, no - I am moving it to config now :)

**Comment by starkers** [comment]
Yes I think README should be plenty TBH (rather than make the role convoluted)

# sample 2.4.1

```
elasticsearch:
  cluster_name: example-elasticsearch
  disable_dynamic_scripting: true
  family:  2.x 
  version: 2.4.1
  index:
    number_of_shards: 3
    number_of_replicas: 1
  instance_count: 3
  instance_type: t2.medium
  instance_monitoring: true
  instance_volume_size: 32
  plugin_config:
    aws:
      cloud_aws_region: eu-west-1
      discovery:
        ec2_tags:
          Stack: example-elasticsearch
        enabled: true
      s3_backup:
        bucket: example-elasticsearch-snapshots
  plugins:
    - plugin_name: lmenezes/elasticsearch-kopf/v.2.1
      plugin_dir:  kopf
    - plugin_name: mobz/elasticsearch-head
      plugin_dir:  head
    - plugin_name: cloud-aws
      plugin_dir:  cloud-aws
    - plugin_name: royrusso/elasticsearch-HQ
      plugin_dir:  hq  
  s3_bucket_name:  example-elasticsearch-snapshots
```

(I like kopf for shard and snapshot administration)

**Comment by sreddel** [comment]
i like this!:+1:

**Comment by wojtek-oledzki** [comment]
@starkers @lobsterdore @racer1680 please check the PR again. It's a breaking change, hence the minor version bump-up to `v1.2`.

File: defaults/main.yml [hunk 2]
```diff
@@ -30,25 +30,22 @@ elasticsearch:
...
     - plugin_name: mobz/elasticsearch-head
-      plugin_dir:  head
+      plugin_dir: head
     - plugin_name: royrusso/elasticsearch-HQ
-      plugin_dir:  HQ
-    - plugin_name: elasticsearch/elasticsearch-lang-mvel/1.4.1
-      plugin_dir: lang-mvel
+      plugin_dir: hq
 
   plugin_config:
```
Comment (lobsterdore) [inline defaults/main.yml]: Should we remove plugin_config and just move s3_backup to the parent hash? Eg.

```YAML
elasticsearch:
  s3_backup:
    enabled: false
    base_path: snapshots
    bucket: changeme
    region: changeme
```

In the future we could have a more configurable backup option but I think that can come later on.

File: templates/2.x/elasticsearch.default.j2 [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# file: elasticsearch/templates/elasticsearch.default.j2
+#
+
+CONF_DIR={{ elasticsearch.paths.configuration }}
+DATA_DIR={{ elasticsearch.paths.data }}
+ES_HOME=/usr/share/elasticsearch
+ES_USER=elasticsearch
+LOG_DIR={{ elasticsearch.paths.logs }}
+MAX_MAP_COUNT=262144
+MAX_OPEN_FILES=65535
```
Comment (lobsterdore) [inline templates/2.x/elasticsearch.default.j2]: These values should come from .limits I think.

File: tests/vars_elasticsearch_1.4.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+# No vars - use role's defaults
```
Comment (lobsterdore) [inline tests/vars_elasticsearch_1.4.yml]: Do we need these extra comments?

File: .travis.yml [hunk 5]
```diff
@@ -43,6 +50,6 @@ script:
     | grep -q 'changed=0.*failed=0'
     && (echo 'Idempotence test: pass' && exit 0)
     || (echo 'Idempotence test: fail' && exit 1)
-    
+
```
Comment (lobsterdore) [inline .travis.yml]: I think we should add a test against the assert tag here.

**Comment by wojtek-oledzki** [comment]
@lobsterdore all addressed

**Comment by lobsterdore** [comment]
LGTM :+1:

**Comment by starkers** [comment]
LGTM LGTM

**Pull request merged**